Add `keepAlive` option to explainer

pythagoraskitty opened the pull request:
We add the option to `keepAlive` a worklet after a `selectURL`/`run` operation. This option will default to false, and if not specified as true, the worklet will close after the operation completes.

jkarlin commented on README.md:
```diff
@@ -113,7 +113,12 @@ The shared storage worklet invocation methods (`addModule`, `run`, and `selectUR
+    *   Options can include:
+        *   `data`, an arbitrary serializable object passed to the worklet. 
+        *   `retain` (defaults to false), a boolean denoting whether the worklet should be retained after it completes work for this call.
+            *   If `retain` is false or not specified, the worklet will shutdown as soon as the operation finishes and subsequent calls to it will fail.
```
nit: prefer keep-alive as I think that's common parlance and more readily understood.

pythagoraskitty replied:
OK, my concern with using "keep-alive" was that it could produce confusion with the "keep-alive" state that happens if the document is destroyed and there are still operations running.  

If you think that this ambiguity is not a problem, then I am happy to change it.

jkarlin replied:
I think this is higher priority in terms of keep-alive term usage, as it's part of the public API. We can rename the "keep-alive worklet" section to something like "Worklets can outlive renderers"

pythagoraskitty replied:
OK, sounds good. Will do, thanks!

pythagoraskitty replied:
Done.

jkarlin commented on README.md:
```diff
@@ -113,7 +113,12 @@ The shared storage worklet invocation methods (`addModule`, `run`, and `selectUR
+        *   `data`, an arbitrary serializable object passed to the worklet. 
+        *   `retain` (defaults to false), a boolean denoting whether the worklet should be retained after it completes work for this call.
+            *   If `retain` is false or not specified, the worklet will shutdown as soon as the operation finishes and subsequent calls to it will fail.
+            *   To continue retaining the worklet throughout multiple calls to `run()` and/or `selectURL()`, each of those calls must include `retain: true` in the `options` dictionary.
```
s/to continue retaining/to keep the worklet alive/

pythagoraskitty replied:
Done.

jkarlin commented on README.md:
```diff
@@ -306,9 +311,68 @@ class ReportingOperation {
 
+### Retaining a worklet for multiple operations
+
+We may wish to run multiple worklet operations from the same context, e.g. we might select a URL and then send one or more aggregatable reports. To do so, we would need to use the `retain: true` option when calling eaching of our worklet operations (except perhaps in the last call, if we have no need to extend the worklet's lifetime beyond that call).
```
editing nit, we seems a bit strange to me. Perhaps, "callers may wish to"? And then below would be something like, s/we could make the calls as follows/As an example:/

pythagoraskitty replied:
Done.

jkarlin commented on README.md:
```diff
@@ -306,9 +311,68 @@ class ReportingOperation {
 ## Keep-alive worklet
 
-After a document dies, the corresponding worklet will be kept alive for maximum two seconds to allow the pending operations to execute. This gives more confidence that the end-of-page operations (e.g. reporting) are able to finish.
+After a document dies, the corresponding worklet (if still alive) will continue to be kept alive for a maximum of two seconds to allow the pending operations to execute. This gives more confidence that the end-of-page operations (e.g. reporting) are able to finish.
```
s/(if still alive)/(if running an operation)/

pythagoraskitty replied:
Done.

jkarlin reviewed (commented):
Overall looks good, some comments:

pythagoraskitty commented:
jkarlin@, please take another look.  Thanks!

xyaoinum commented on README.md:
````diff
@@ -306,9 +311,66 @@ class ReportingOperation {
+register('report', ReportOperation);
+```
+
+## Worklets can outlive renderers
````
I think it’s more natural to say that "Worklets can outlive the associated Document", as I’m not sure if "renderer" is a standard term.

xyaoinum reviewed (approved):
LGTM % a nit

pythagoraskitty commented:
I thought that I already changed all of the instances.  Where did I miss?

In articular, I already changed the section heading of "Keep-alive worklet" to "Worklets can outlive the associated document"

jkarlin commented:
> 

My bad, I missed that change. LGTM
